modular_skyrat/master_files/code/modules/client/preferences/mutant_parts.dm

```diff
@@ -392,95 +392,199 @@
+	default_accessory_type = /datum/sprite_accessory/synth_head/default
+	should_generate_icons = TRUE
+	generate_icons = TRUE
+	crop_area = list(11, 22, 21, 32) // We want just the head.
+	greyscale_color = DEFAULT_SYNTH_PART_COLOR
 
-/datum/preference/tri_color/ipc_head
-	category = PREFERENCE_CATEGORY_SECONDARY_FEATURES
+/datum/preference/choiced/mutant_choice/synth_head/generate_icon_state(datum/sprite_accessory/sprite_accessory, original_icon_state)
+	// If this isn't the right type, we have much bigger problems.
+	var/datum/sprite_accessory/synth_head/head = sprite_accessory
+	return "[original_icon_state]_head[head.dimorphic ? "_m" : ""]"
+
+/datum/preference/choiced/mutant_choice/synth_head/is_part_enabled(datum/preferences/preferences)
+	return TRUE
+
+/datum/preference/choiced/mutant_choice/synth_head/compile_constant_data()
+	var/list/data = ..()
+
+	data[SUPPLEMENTAL_FEATURE_KEY] = "ipc_head_color"
+
+	return data
+
+/datum/preference/color/mutant/synth_head
+	category = PREFERENCE_CATEGORY_SUPPLEMENTAL_FEATURES
 	savefile_identifier = PREFERENCE_CHARACTER
 	savefile_key = "ipc_head_color"
-	relevant_mutant_bodypart = "ipc_head"
-	type_to_check = /datum/preference/toggle/mutant_toggle/ipc_head
+	relevant_mutant_bodypart = MUTANT_SYNTH_HEAD
+
+// Synth Hair Opacity
+
+/datum/preference/toggle/mutant_toggle/synth_hair_opacity
+	savefile_key = "feature_synth_hair_opacity_toggle"
+	relevant_mutant_bodypart = MUTANT_SYNTH_HAIR
+
+/datum/preference/numeric/synth_hair_opacity
+	category = PREFERENCE_CATEGORY_SECONDARY_FEATURES
+	savefile_identifier = PREFERENCE_CHARACTER
+	savefile_key = "feature_synth_hair_opacity"
+	relevant_mutant_bodypart = MUTANT_SYNTH_HAIR
+	maximum = 255
+	minimum = 40 // Any lower, and hair's borderline invisible on lighter colours.
+
+/datum/preference/numeric/synth_hair_opacity/create_default_value()
+	return 255
+
+/datum/preference/numeric/synth_hair_opacity/is_accessible(datum/preferences/preferences)
+	var/passed_initial_check = ..(preferences)
+	var/allowed = preferences.read_preference(/datum/preference/toggle/allow_mismatched_parts) && preferences.read_preference(/datum/preference/toggle/mutant_toggle/synth_hair_opacity)
+	return passed_initial_check || allowed
+
+/**
+ * Actually applied. Slimmed down version of the logic in is_available() that actually works when spawning or drawing the character.
+ *
+ * Returns if feature is visible.
+ *
+ * Arguments:
+ * * target - The character this is being applied to.
+ * * preferences - The relevant character preferences.
+ */
+/datum/preference/numeric/synth_hair_opacity/proc/is_visible(mob/living/carbon/human/target, datum/preferences/preferences)
+	var/species_type = preferences.read_preference(/datum/preference/choiced/species)
+	var/datum/species/species = new species_type
+
+	var/species_available = (savefile_key in species.get_features())
+
+	var/overriding = preferences.read_preference(/datum/preference/toggle/allow_mismatched_parts)
```

Comment:
I feel like this should be checked first, before initializing the species, to avoid the overhead for all the ones that won't use this as a synthetic.

Reply:
Feels like the comments that I made on this file weren't properly addressed, when it came to requested changes.

Reply:
The single emissive toggle base class was done, I just forgot to update the path and remove old code.

I seemed to have forgotten about the hair opacity `is_visible` proc, and I've instead optimized it to use initial along with the global list of instantiated species, along with what you suggested, and gave the normal mutant `is_visible` proc the same code too.